my_enumerable.rb

```diff
@@ -0,0 +1,185 @@
+      idx += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    select_arr = []
+    my_each { |value| select_arr << value if yield(value) }
+    select_arr
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      to_a.my_each { |value| return false if yield(value) == false }
+      return true
+    elsif arg.nil?
+      to_a.my_each { |value| return false if value.nil? || value == false }
+    elsif !arg.nil? && (arg.is_a? Class)
+      to_a.my_each { |value| return false unless [value.class, value.class.superclass].include?(arg) }
+    elsif !arg.nil? && arg.class == Regexp
+      to_a.my_each { |value| return false unless arg.match(value) }
+    else
+      to_a.my_each { |value| return false if value != arg }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      to_a.my_each { |idx| return true if yield(idx) }
+      return false
+    elsif arg.nil?
+      to_a.my_each { |idx| return true if idx }
+    elsif !arg.nil? && (arg.is_a? Class)
+      to_a.my_each { |idx| return true if [idx.class, idx.class.superclass].include?(arg) }
+    elsif !arg.nil? && arg.class == Regexp
+      to_a.my_each { |idx| return true if arg.match(idx) }
+    else
+      to_a.my_each { |idx| return true if idx == arg }
+    end
+    false
+  end
+
+  def my_none?(arg = nil, &block)
+    !my_any?(arg, &block)
+  end
+
+  def my_count(arg = nil)
+    count = 0
+    if arg.nil? && block_given?
+      my_each { |value| count += 1 if yield(value) }
+    elsif arg.nil?
+      my_each { |_value| count += 1 }
+    else
+      my_each { |value| count += 1 if value == arg }
+    end
+  end
+
+  def my_map(proc = nil)
+    return to_enum unless block_given?
+
+    result = []
+    proc.nil? ? my_each { |idx| result << yield(idx) } : my_each { |idx| result << proc.call(idx) }
+    result
+  end
+
+  def my_inject(arg1 = nil, arg2 = nil)
+    array = is_a?(Array) ? self : to_a
+    symbol_ = arg1 if arg1.is_a?(Symbol) || arg1.is_a?(String)
+    actual_item = arg1 if arg1.is_a? Integer
+
+    if arg1.is_a?(Integer)
+      if arg2.is_a?(Symbol) || arg2.is_a?(String)
+        symbol_ = arg2
+      elsif !block_given?
+        raise "#{arg2} is not a symbol nor a string"
+      end
+    elsif arg1.is_a?(Symbol) || arg1.is_a?(String)
+      raise "#{arg2} is not a symbol nor a string" if !arg2.is_a?(Symbol) && !arg2.nil?
+
+      raise "undefined method `#{arg2}' for :#{arg2}:Symbol" if arg2.is_a?(Symbol) && !arg2.nil?
+    end
+
+    if symbol_
+      array.my_each { |choice| actual_item = actual_item ? actual_item.send(symbol_, choice) : choice }
+    elsif block_given?
+      array.my_each { |choice| actual_item = actual_item ? yield(actual_item, choice) : choice }
+    else
+      raise 'no block given'
+    end
+    actual_item
+  end
+end
+
+# rubocop: enable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+def multiply_els(arr)
+  p arr.my_inject(1) { |d, v| d * v }
+end
```

Comment:
- [ ] should have working `multiply_els` that uses `my_inject` as described on Odin ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/multiply_els.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))

#### multiply_else

- [ ] accepts an array as an argument and multiplies all the elements of the array together using #my_inject

```ruby
multiply_els [2, 4, 5] => 40
```
- multiply_els is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/95802136-d5c6f100-0cb9-11eb-84e2-a77013038dc0.png)